### DISCUSSION TITLE
Disguises that fool Character AI

### DISCUSSION THREAD

**Discussion by Ruzihm** [FakeFishGames/Barotrauma#7435] [Ideas and suggestions]
It would be interesting if the Character AI, instead of checking the actual team membership of other characters to determine how they should be treated, they used the apparent team membership of the other character, which would be their actual team membership if they are not disguised, or the team membership of the disguise if they are wearing a disguise.

This leads to fun situations like the following. (using pirate mission as just an example):

Kill one pirate, use their ID card for a disguise, and then board their ship and sabotage it.
Stun one pirate, handcuff them, put a player-team ID and a face covering on them, let them draw the attacks of their crew mates.

There are some design decisions to make about this especially regarding if/when a disguise should not be effective against the AI. Should a character "expose" the disguise under certain conditions such as witnessing hostile action?
Should a character immediately override the disguise if they witness the disguise being put on?
Once a character exposes a disguise, should the disguise be exposed for other characters?
If a particular ID card is used as a disguise then gets exposed, can that ID card work for a disguise again for that session? If yes, under what conditions?

Once implemented at a basic level, this "apparent team" behavior could be extended to new types of mission that gives the player an ID to sneak aboard a vessel. Or, an extension of the terrorist mission where the terrorist(s) on board put on disguises to fool the players and their AI crewmates (which would override the "no disguises" server option, just for these characters, ideally). With the disguised terrorist mission, if it's multiple terrorists, it might be best if they always ignored each others' disguise, but maybe not for extra chaos.

**Comment by StruggleDuck** [comment]
yes please

**Comment by Rosentti** [comment]
This is how I thought disguises worked and I tried to go into a hostile outpost with one to grab missions. It did not end well.

**Reply by JoebamaCrump** [reply]
I tried the exact same thing

**Comment by gael-vanderlee** [comment]
I'd love to see this implemented
